src/main/java/com/bookstore/model/Order.java

```diff
@@ -0,0 +1,62 @@
+package com.bookstore.model;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Set;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE orders SET is_deleted=true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Entity
+@EqualsAndHashCode
```

Comment:
It seems **id** is the only field we need for Equals And HashCode in Order class.
It is impossible to have Order without id, so, id is a business key for Order.